upstairs/src/block_req.rs

```diff
@@ -57,16 +101,43 @@ impl BlockRes {
     ) {
         // XXX this eats the result!
         let _ = self
-            .0
+            .tx
             .take()
             .expect("sender was populated")
             .send(BlockReqReply { buffer, result });
     }
+
+    /// Add this job to the given backpressure counters
+    ///
+    /// The backpressure counters are stored locally and decremented when this
+    /// `BlockRes` is dropped, so we automatically keep track of outstanding
+    /// work without human intervention.
+    fn bind(&mut self, op: &BlockOp, bp: Arc<BackpressureCounters>) {
+        // Count this job as 1 job and some number of bytes
+        //
+        // (we only count write bytes because only writes return immediately)
+        assert!(self.backpressure_guard.is_none());
+        let bytes = match op {
+            BlockOp::Write { data, .. } => data.len() as u64,
```

Comment:
does it make sense to include `WriteUnwritten` here? That should never come from the Guest, except through scrubbing.

Reply:
My intuition is "no", because `WriteUnwritten` operations don't get to use the fast-ack optimization, so they should get "natural" backpressure (in that we'll only reply to the guest once the operation is done*).  That's how it's handled in the current code, so my (weakly-held) inclination is to leave it be.

*with some caveats related to #1167, but that's neither here nor there